aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -791,6 +780,23 @@ void TensorIterator::mark_outputs() {
   }
 }
 
+void TensorIterator::mark_resize_outputs(const TensorIteratorConfig& config) {
+  if (config.static_shape_.has_value()) {
+    return;
+  }
+  for (int i = 0; i < num_outputs_; i++) {
+    const auto& output = operands_[i].tensor;
+    if (output.defined() && !output.sizes().equals(shape_)) {
+      if (config.resize_outputs_ && !operands_[i].is_read_write) {
+        operands_[i].will_resize = true;
+        continue;
+      }
+      TORCH_CHECK(is_reduction_,  "output with shape ", output.sizes(), " doesn't match the broadcast shape ",
```

Comment:
just curious - why is_reduction case is fine? maybe add a comment? (I know this logic predated this diff, but still)